pkg/trie/triedb/node_storage.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package triedb
+
+import (
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/trie/triedb/codec"
+	"github.com/gammazero/deque"
+)
+
+// StorageHandle is a pointer to a node contained in `NodeStorage`
+type StorageHandle struct{ int }
```

Comment:
don't think this needs to be exported and the type could be simpler.
```suggestion
type storageHandle int
```

Reply:
Yes! changed!